src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -991,21 +1019,39 @@ public boolean equals(Object obj) {
                 } else if (event instanceof RefUpdated) {
                     RefUpdated refUpdated = (RefUpdated)event;
                     if (isServerInteresting(event) && p.isInteresting(refUpdated.getRefUpdate().getProject(),
-                                                                      refUpdated.getRefUpdate().getRefName(), null)) {
+                            refUpdated.getRefUpdate().getRefName(), null, envVars)) {
                         logger.trace("According to {} the event is interesting.", p);
                         return true;
                     }
                 }
             } catch (PatternSyntaxException pse) {
                 logger.error(MessageFormat.format("Exception caught for project {0} and pattern {1}, message: {2}",
-                       new Object[]{job.getName(), p.getPattern(), pse.getMessage()}));
+                        new Object[]{job.getName(), p.getPattern(), pse.getMessage()}));
             }
         }
         logger.trace("Nothing interesting here, move along folks!");
         return false;
     }
 
     /**
+     * Checks if the current system env vars are still the same and updates our local copy if necessary
+     */
+    private void updateEnvVarsIfNecessary() {
+        EnvironmentVariablesNodeProperty envVarsProp =
+                Jenkins.getActiveInstance().getGlobalNodeProperties().get(EnvironmentVariablesNodeProperty.class);
+        EnvVars globalEnvVars = envVarsProp.getEnvVars();
+        int newSystemEnvVarsHash = EnvVars.masterEnvVars.hashCode();
+        int newGlobalEnvVarsHash = globalEnvVars.hashCode();
+        if (newSystemEnvVarsHash != systemEnvVarsHash || newGlobalEnvVarsHash != globalHashVarsHash) {
+            this.envVars.clear();
```

Comment:
very good point. It extends `TreeMap` so unfortunately its not. Is there a performance penalty to consider or is it enough to make it `synchronized`?